docs/contributing/testing-guidelines/integration-testing.md

```diff
@@ -0,0 +1,267 @@
+# Integration Testing
+
+## Introduction
+
+The purpose of this guide is to motivate developers to adopt integration testing best practices with Autoware by explaining how to write, run,
+and evaluate the results of integration tests.
+
+## Quick reference
+
+1. [colcon](https://github.com/ros2/ros2/wiki/Colcon-Tutorial) is used to build and run tests.
+2. [launch testing](https://github.com/ros2/launch/tree/master/launch_testing) launches nodes and runs tests.
+3. [Testing guidelines](index.md) describes the different types of tests performed in Autoware and links to the corresponding guidelines.
+
+## Integration testing
+
+An integration test is defined as the phase in software testing where individual software
+modules are combined and tested as a group. Integration tests occur after unit tests, and before
+validation tests.
+
+The input to an integration test is a set of independent modules that have been unit tested. The set
+of modules is tested against the defined integration test plan, and the output is a set of
+properly integrated software modules that is ready for system testing.
+
+## Value of integration testing
+
+Integration tests determine if independently developed software modules work correctly
+when the modules are connected to each other. In ROS 2, the software modules are called
+nodes. As a special case, testing a single node can be referred to as component testing.
+
+Integration tests help to find the following types of errors:
+
+- Incompatible interactions between nodes, such as non-matching topics, different message types, or incompatible QoS settings
+- Edge cases that were not touched by unit testing, such as a critical timing issue, network communication delays, disk I/O failures, and many other problems that can occur in production environments
+- Issues that can occur while the system is under high CPU/memory load, such as `malloc` failures. This can be tested using tools like `stress` and `udpreplay` to test the performance of nodes with real data.```
+
+With ROS 2, it is possible to program complex autonomous-driving applications with a large number
+of nodes. Therefore, a lot of effort has been made to provide an integration-test framework that
+helps developers test the interaction of ROS 2 nodes.
+
+## Integration-test framework
+
+A typical integration-test framework has three parts:
+
+1. A series of executables with arguments that work together and generate outputs
+2. A series of expected outputs that should match the output of the executables
+3. A launcher that starts the tests, compares the outputs to the expected outputs, and determines if the test passes
+
+In Autoware, we use the [launch_testing](https://github.com/ros2/launch/tree/master/launch_testing) framework.
+
+### Smoke tests
+
+Autoware has a dedicated API for smoke testing
+
+To use this framework, in `package.xml` add:
+
+```{xml}
```

Comment:
Could you remove all `{}` from the language of code blocks?
It prevents highlighting in some environments.

``````suggestion
```xml
``````

Reply:
Done 08fdaba9012bcbf64c5117f139f993cb87da5854